owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -347,26 +367,79 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
 
     /**
      * OAuth step 1: Get authorization code
+     * Firstly, try the OAuth authorization with Open Id Connect, checking whether there's an available .well-known url
+     * to use or not
      */
-    private fun startOAuthorization() {
+    private fun startOIDCOauthorization() {
         server_status_text.run {
             setCompoundDrawablesWithIntrinsicBounds(R.drawable.progress_small, 0, 0, 0)
             text = resources.getString(R.string.oauth_login_connection)
         }
 
-        mAuthService = OAuthUtils.createAuthorizationService(this)
+        val retrieveConfigurationCallback =
+            RetrieveConfigurationCallback { serviceConfiguration, exception ->
+                if (exception != null) {
+                    Timber.e(exception, "OIDC failed, try with normal OAuth");
+                    startNormalOauthorization();
+                } else if (serviceConfiguration != null) {
+                    oidcSupported = true;
+                    performGetAuthorizationCodeRequest(serviceConfiguration)
+                    authorizationServiceConfiguration = serviceConfiguration
+                }
+            }
+
+        OAuthUtils.buildOIDCAuthorizationServiceConfig(
+            this,
+            serverBaseUrl,
+            retrieveConfigurationCallback
+        )
+    }
+
+    /**
+     * OAuth step 1: Get authorization code
+     * If OIDC is not available, falling back to normal OAuth
+     */
+    private fun startNormalOauthorization() {
+        val retrieveConfigurationCallback =
+            RetrieveConfigurationCallback { serviceConfiguration, exception ->
+                if (exception != null) {
+                    Timber.e(exception, "Normal OAuth failed");
+                    updateOAuthStatusIconAndText(exception);
```

Comment:
Same here, could you remove every `;` from this file?

Reply:
![travolta-despistado-300x242](https://user-images.githubusercontent.com/15261363/79972603-833d1280-8496-11ea-9374-d19ec63fc77b.gif)